app/src/main/res/values/strings.xml

```diff
@@ -21,6 +22,10 @@
     <string name="error_login_unknown">Unknown error logging in.</string>
     <string name="error_login_network">Network error logging in.</string>
 
+    <!-- logout -->
+    <string name="logout_dialog_title">Are you sure you want to log out?</string>
+    <string name="logout_dialog_message">You will need to log in again to enter or manage records after logging out. Logging in will require Internet access. Are you sure you want to log out now?</string>
```

Comment:
No, logging out does not affect the local database of records. They'll stay there until the user logs in again, under any account. The logout warning prompt is there to make sure they don't accidentally log out while in the field without Internet access, leaving them unable to use the app again until they get back on a network somewhere.

It is necessary to upload/clear the local records before updating the schema; I take it that's the other warning message you're looking at. There's no risk of record loss with the update. The old records need to be cleared out to deal with the class reloading cleanly.